packages/block-library/src/list/transforms.js

```diff
@@ -208,6 +208,13 @@ const transforms = {
 				} );
 			},
 		},
+		{
+			type: 'block',
+			blocks: [ 'core/table-of-contents' ],
+			transform: () => {
+				return createBlock( 'core/table-of-contents' );
+			},
+		},
```

Comment:
I think of it as serving a similar function to the transforms for the Navigation Link that let you convert a Page Link to a Category Link (and etc.)... obviously, the block just gets replaced with little to nothing kept from the original block, but the purpose is to make replacing the block easier. If you convert a Table of Contents to a static list, but then want to go back later, a transform is more convenient than deleting the List and inserting a ToC. Or at least that's the idea.

(In the future, some simple stuff like background/text color from the List could be transferred over, but I decided against adding those to the ToC right now to avoid making this PR even bigger.)